### ISSUE TITLE
consider bridging hls.js

### ISSUE THREAD

**Issue by aronallen** [videojs/videojs-contrib-hls#437]
Hi,

I have successfully branched the latest version of videojs (5.2.0) and made a tech bridge to [/dailymotion/hls.js](/dailymotion/hls.js)
The documentation on tech extensions is quite sparse, so the only solution I could find was to build my tech with the videojs project, since it has many dependencies.

I generally find HLS.JS more stable than videojs-contrib-hls, though it has problems recognizing live streams, so I made an option that you can pass in `hlsjs : {live : true}` to the videojs config object and force the duration to Infinity. I have also provided an example in the sandbox called hls.html.example

I think the open source community would be better of with a stable HLS implantation, independent from a specific player, and the different players then provided bridges for those. By joining forces with HLS js, and having this project focused on bridging, we could all reap the benefits.

you can see my work here [/aronallen/video.js](/aronallen/video.js)

**Comment by tchakabam** [comment]
generally totally agree on the idea :+1:

**Comment by dmlap** [comment]
I spoke with @mangui a little bit before hls.js was released and we both wanted to figure out how to combine efforts but it's a bit tricky when you get down to the details. One offshoot of that discussion was spinning off [mux.js](https://github.com/videojs/mux.js) instead of building equivalent transmuxers for each project independently. If both projects shared that piece of infrastructure that would be pretty cool and a lot less duplicated work.

Merging the playback engines themselves is harder because porting functionality is hard. My impression is hls.js is further along with stream/platform compatibility and has some cool features like redundant streams that contrib-hls lacks. I believe contrib-hls's live support is deeper (though there are still plenty of problem streams) and we have some features hls.js is missing, like support for AES-128 HLS encryption, timed ID3 metadata, and CEA-608 captions.

So in conclusion: awesome work hooking up hls.js to video.js! I'm with you completely that we need to figure out how to work together better. Maybe that means spinning off more parts of hls.js and this project into pieces we can work together on? I'm open to any ideas on the right places to cut things up. @mangui any thoughts to add?

**Comment by aronallen** [comment]
Sounds great you have similar ideas, I really think this whole streaming thing is a mess right now, too many formats, protocols, and implementations, add the diversity of content decryption modules and you are deep down a rabbit hole. If we could somewhat limit this Cambrian explosion by aligning projects it could be great. The guys at Canal Plus are also working on some cool [DASH](/canalplus/rx-player) technology, built on top of reactive extensions, a reactive approach seems to be more sane in the highly dangerous and asynchronous world of adaptive streaming, but developers are in general less familiar to reactive approaches, I am myself only getting familiar with it, my dream is to build a rx based HLS player, and a well tested mux.js would ease my efforts.

It could be cool to develop some small AI that was specialized in parsing m3u8 manifests, there are so many slight variations in them, the variations are often obvious to humans, but a regular parser seems to very sensitive. I am no AI expert, but I imagine it should be possible to make something work, in the term space of HLS manifests.

In addition, I would like to know how to make my tech a propper plugin like this one. it seems in videojs 5 a lot of objects are private and I can't figure out how to hook into them dynamically, also I don't quite know the difference between a component and a tech, if someone had the time to answer my questions on gitter, I would be very grateful, and I could release the plugin independently, instead of forking videojs.

**Comment by sgavali** [comment]
Is MSE support varies between Android release ? I tried this plugin for a issue https://github.com/videojs/videojs-contrib-hls/issues/424 I was facing and it worked on Android 4.4.x but crashed on Android 4.3

**Comment by FDiskas** [comment]
To many issues on dailymotions LIB. This one works better. For example LG G3 phone is not playing videos on dailymotions but works on this

**Comment by zuzzurro** [comment]
I see you mention support for CEA-608 captions as a feature you have as opposed to hls.js. I am trying the player with a captioned stream but I see nothing showing up. Is the support really there?

**Comment by imbcmdth** [comment]
@zuzzurro We have continued to work on caption support and recently fixed some bugs related to captions. If you still have problems with the latest builds it would help us out if you could provide some samples. One of the most difficult parts of 608/708 development is finding samples to test against!

The only remaining limitation that I know of is the lack of official "paint-on" caption mode support.

**Comment by zuzzurro** [comment]
I would love to try it. Don't you have a demo page where I can easily load and test my videos? Flashls demo pages are very convenient in this respect.

**Comment by zuzzurro** [comment]
I have two videos, coming from two different encoder brands. Flashls is able to decode the captions fine in one video while can't decode the other ones. I'm pretty sure the files are ok as both VLC and jwplayer show the caption tracks. I would love to compare your demux implementation in order to see its behavior.

**Comment by jimmywarting** [comment]
Bump, updates?

**Comment by chriswiggins** [comment]
@jimmywarting try [peer5/videojs-contrib-hls.js](https://github.com/Peer5/videojs-contrib-hls.js/)

**Comment by gesinger** [comment]
@jimmywarting , sorry that we haven't had any updates in a while. This has actually come up multiple times, and we think the best overall idea is to have the projects working closer together to develop shared components such that the open source community doesn't have too many different codebases doing similar things (with different bugs/features in each).

This is still a great idea, and we should start pushing towards the goal. I think the best way to start may be by extracting modules for specific functionality. This is something we are looking into doing more in the near future (and have done with certain aspects, e.g., https://github.com/videojs/m3u8-parser and https://github.com/videojs/aes-decrypter ). Once we have distinct modules for critical features, and different implementations can combine them in simple ways, it may promote more reuse among the community.

We also opened https://github.com/dailymotion/hls.js/issues/477 and while I haven't had a chance to look into it too much yet, I think it may be another great place to work from.

**Comment by forbesjo** [comment]
Closing since the dailymotion issue has been closed
